### PR TITLE
GH-39170: [Java] Improve error message explaining why TestTls might fail

### LINKED ISSUE #39170: [Java] Improve error message when Flight Core TestTls fails
### Describe the enhancement requested

As discussed on the mailing list, when we forgot `git submodule update --init --recursive`, `cert0.pem` and other files are not present in `testing/data` folder.
It means `TestTls` fails, but it's not obvious to find why related to https://arrow.apache.org/docs/dev/developers/java/building.html#building.

I propose to display a "dev friendly" message ;)

### Component(s)

Java

### PR DESCRIPTION
**Pull request by jbonofre**
### Are there any user-facing changes?

* Closes: #39170

### CODE REVIEW

File: java/flight/flight-core/src/test/java/org/apache/arrow/flight/FlightTestUtil.java [hunk 1]
```diff
@@ -47,7 +48,11 @@ static Path exampleTlsRootCert() {
 
   static List<CertKeyPair> exampleTlsCerts() {
     final Path root = getFlightTestDataRoot();
-    return Arrays.asList(new CertKeyPair(root.resolve("cert0.pem").toFile(), root.resolve("cert0.pkcs1").toFile()),
+    final Path cert0Pem = root.resolve("cert0.pem");
+    if (!Files.exists(cert0Pem)) {
+      throw new RuntimeException(cert0Pem + " doesn't exist. Take a look on https://arrow.apache.org/docs/dev/developers/java/building.html#building to initialize.");
```
Comment (lidavidm) [inline java/flight/flight-core/src/test/java/org/apache/arrow/flight/FlightTestUtil.java:53]:
```suggestion
      throw new RuntimeException(cert0Pem + " doesn't exist. Make sure submodules are initialized (see https://arrow.apache.org/docs/dev/developers/java/building.html#building)");
```

Comment (lidavidm) [reply java/flight/flight-core/src/test/java/org/apache/arrow/flight/FlightTestUtil.java:53]: Something like this, in case the user already knows git?

**Review by lidavidm — APPROVED** [review]
Thanks!

File: java/flight/flight-core/src/test/java/org/apache/arrow/flight/FlightTestUtil.java (hunk 1 above)
Comment (jbonofre) [reply java/flight/flight-core/src/test/java/org/apache/arrow/flight/FlightTestUtil.java:53]: Good idea. Let me do the change. Thanks again !

Comment (jbonofre) [reply java/flight/flight-core/src/test/java/org/apache/arrow/flight/FlightTestUtil.java]: Done, thanks again for the suggestion.

**Comment by jbonofre** [comment]
@lidavidm I updated the PR according to your suggestion. Thanks again. This PR is the first one of several coming from me 😄

**Review by lidavidm — APPROVED** [review]

**Pull request merged**